### ISSUE TITLE
Images are exposed as text to accessibility API until loaded

### ISSUE THREAD

**Issue by jnurthen** [aFarkas/lazysizes#653]
When browsing a page with lazy loaded images using a screen reader the images are not reported as graphics until they are loaded. In the example below
```
    <h1>Foobar Flowershop</h1>
    <!-- Images Start -->
    <img src="images/flower1.jpg" alt="wibble"/>
    <img src="images/flower2.jpg" alt="too many flowers"/>
    <img data-src="images/flower3.png" class="lazyload" alt="flower again"/>
    <img data-src="images/flower4.jpg" class="lazyload" alt="flower"/>
    <img data-src="images/flower5.jpg" class="lazyload" alt="another flower"/>
    <!-- Images End -->
    <script src="lazysizes.min.js" async></script>
```
I get the following from the accessibility API
<img width="1320" alt="Screen Shot showing accessibility tree from FF with 2 images and 3 text objects" src="https://user-images.githubusercontent.com/1724479/57396678-be5c0e00-717f-11e9-8333-c86432928e38.png">

According to the HTML spec this is the correct behaviour and images without an src should expose the alt as text content. Problem is I can't see what would ever force the image to load when using a screen reader and populate the src attribute to make the image a graphic. 

Potential solution is to add role="img" to the image until the script runs. This seems to correct the behaviour.

**Comment by aFarkas** [comment]
What tool is this?

My in browser too gives me the following tree:
<img width="1020" alt="Bildschirmfoto 2019-05-08 um 22 15 52" src="https://user-images.githubusercontent.com/188254/57406784-ee3bfd80-71e1-11e9-8748-fd9e53d8edc8.png">

> Potential solution is to add `role="img"` to the image until the script runs.

Yes this was also my thought.

**Comment by jnurthen** [comment]
@aFarkas yeah it looks fine in Chrome and Safari in my testing too. I was testing in Firefox. The tool is just the Accessibility Inspector which is part of standard Dev Tools (for at least a few versions).

According to spec the FF behaviour is correct.

**Comment by hiwelo** [comment]
:+1: to add `role="img"` to be as close as possible to the spec and to provide a good experience to users of assistive technologies on Windows.
(one the most common configuration on Windows is NVDA + Firefox if I'm right)

**Comment by aFarkas** [comment]
other solution would be to add `src=data:....`

**Comment by hiwelo** [comment]
Like `<img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" alt="">` to display a transparent 1x1 gif?

**Comment by aFarkas** [comment]
Yes, this was actually the old favored way. I removed this from all of my examples because browsers do not support progressive image loading and some browsers also tend to swap the images more slowly.

One thing to check would be wether the change of the virtual cursor also scrolls the page, then this is already fixed by default.

**Comment by hiwelo** [comment]
Even if the virtual cursor also scrolls the page, all images outside of the viewport and not already downloaded will not appear within the list of images.
It can have an impact on meaningful images being invisible for users of assistive technologies using Web Rotor or similar systems to navigate by listings instead of moving the cursor 🙂
